### PR TITLE
16521- code scanner widget - outline

### PR DESCRIPTION
**Pull request by harshilp24**
Outline for review

### CODE REVIEW

File: .github/drafts/16521/16521-Code-Scanner-Outline.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+| Property        | Type                 | Description                                                                                                         | Code Snippet                |
+|-----------------|----------------------|---------------------------------------------------------------------------------------------------------------------|-----------------------------|
+| Text            | Formatting           | Sets the label text of the widget.                                                                                  | -                           |
+| Visible         | Binding & Formatting | Controls widget's visibility on the page. When turned off, the widget will not be visible when the app is published | {{CodeScanner1.isVisible}}  |
+| Disabled        | Binding & Formatting | Disables input to the widget. The widget will remain visible to the user but a user input will not be allowed.      | {{CodeScanner1.isDisabled}} |
+| Animate Loading | Formatting           | Controls the loading of the widget.                                                                                 | -                           |
```
Comment (jnikhila) [inline .github/drafts/16521/16521-Code-Scanner-Outline.md:27]: Could you please add a corss-reference to the common properties section available on widget landing page for this?

File: .github/drafts/16521/16521-Code-Scanner-Outline.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+**//cross-link common properties** 
+
+### Text
+
+### Visible
+
+### Disabled
+
+ ### Animate Loading
```
Comment (jnikhila) [inline .github/drafts/16521/16521-Code-Scanner-Outline.md:41]: No need to explain the property here. Just adding a cross-reference to Animate loading property description in properties table above would suffice.

File: .github/drafts/16521/16521-Code-Scanner-Outline.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+|-----------------|----------------------|---------------------------------------------------------------------------------------------------------------------|-----------------------------|
+| Text            | Formatting           | Sets the label text of the widget.                                                                                  | -                           |
+| Visible         | Binding & Formatting | Controls widget's visibility on the page. When turned off, the widget will not be visible when the app is published | {{CodeScanner1.isVisible}}  |
+| Disabled        | Binding & Formatting | Disables input to the widget. The widget will remain visible to the user but a user input will not be allowed.      | {{CodeScanner1.isDisabled}} |
+| Animate Loading | Formatting           | Controls the loading of the widget.                                                                                 | -                           |
+| Tooltip         | Formatting           | It sets a tooltip for the widget. You can add hints or extra information about the required input from the user.    | -                           |
```
Comment (jnikhila) [inline .github/drafts/16521/16521-Code-Scanner-Outline.md:28]: Same for tooltip as it's a common property used across widgets.

File: .github/drafts/16521/16521-Code-Scanner-Outline.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+### Text
+
+### Visible
+
+### Disabled
+
+ ### Animate Loading
+
+### Tooltip
```
Comment (jnikhila) [inline .github/drafts/16521/16521-Code-Scanner-Outline.md:43]: No need to explain the property here. Just adding a cross-reference to tooltip property description in properties table above would suffice.

File: .github/drafts/16521/16521-Code-Scanner-Outline.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+| Icon           | Sets an icon to be included in the input field.        |
+| Position       | Sets the label position of the widget.                 |
+| Placement      | Sets the label alignment of the widget.                |
+| Button Color   | Allows you to set color for the button.                |
+|  Border Radius | Allows you to define curved corners.                   |
+| Box Shadow     | Allows you to choose from the available shadow styles. |
+
+## Conclusion
+## Cross Reference Section
+
```
Comment (jnikhila) [inline .github/drafts/16521/16521-Code-Scanner-Outline.md:71]: Could you please add the reference links that you think will be added here? Also, please add a cross-reference to troubleshooting section.

Comment (harshilp24) [reply .github/drafts/16521/16521-Code-Scanner-Outline.md:70]: Made necessary changes.

**Comment by harshilp24** [comment]
@jnikhila removed the cross-link hint. Can you have a look and merge it?

**Comment by dhruvikn** [comment]
Not sure if this is relevant to the outline, but can we add a table for supported formats as done here in the PR? I forgot to include this in the draft. 
https://github.com/appsmithorg/appsmith/pull/15990#issue-1337183361

Else, the outline looks good to me.

**Comment by harshilp24** [comment]
> Not sure if this is relevant to the outline, but can we add a table for supported formats as done here in the PR? I forgot to include this in the draft. [appsmithorg/appsmith#15990 (comment)](https://github.com/appsmithorg/appsmith/pull/15990#issue-1337183361)
> 
> Else, the outline looks good to me.

we can add it in the use case section

**Review by dhruvikn — APPROVED** [review]

**Review by jnikhila — APPROVED** [review]

**Pull request merged**